### PR TITLE
Wrap/unwrap match arms

### PR DESCRIPTION
**Pull request by marcusklaas**
Closes https://github.com/rust-lang-nursery/rustfmt/issues/390.

### CODE REVIEW

**Comment by marcusklaas** [comment]
@nikomatsakis mentioned in one of the rustfmt-rust PRs that we should consider wrapping either all arms in blocks or none to improve consistency within a match. Looking at the current results, I think this makes sense. It's not implemented in this PR, but I think we should have an option for this at least.

File: tests/system.rs [hunk 1]
```diff
@@ -249,8 +249,10 @@ fn handle_result(result: HashMap<String, String>,
 fn get_target(file_name: &str, target: Option<&str>, write_mode: WriteMode) -> String {
     let file_path = Path::new(file_name);
     let (source_path_prefix, target_path_prefix) = match write_mode {
-        WriteMode::Coverage => (Path::new("tests/coverage-source/"),
-                                "tests/coverage-target/"),
```
Comment (nrc) [inline tests/system.rs]: I wonder if we should make an exception to the multi-line rule for literals some how? E.g., a multi-line if or match looks better in a block, but I feel like here the original is better. Just a thought for later improvement.

File: tests/target/match.rs [hunk 2]
```diff
@@ -13,13 +13,12 @@ fn foo() {
             foo()
         }
         // Perhaps this should introduce braces?
```
Comment (nrc) [inline tests/target/match.rs]: Can remove this comment now that we do this.

File: tests/target/match.rs [hunk 3]
```diff
@@ -268,14 +276,15 @@ fn issue496() {
...
                     kind: StmtKind::Expr {
                         scope: cx.tcx.region_maps.node_extent(id),
                         expr: expr.to_ref(),
                     },
-                }))),
+                })))
+            }
         }
     }
 }
```
Comment (nrc) [inline tests/target/match.rs]: There might be one already, some where, but if not, could you add a test where the body fits on one line in a block, but is semi-colon terminated - in this case, the block should not be removed.

File: src/expr.rs [hunk 4]
```diff
@@ -896,65 +902,53 @@ impl Rewrite for ast::Arm {
             line_start += offset.width();
         }
 
+        let body = match **body {
+            ast::Expr { node: ast::ExprBlock(ref b), .. } if !is_unsafe_block(b) &&
+                                                             is_simple_block(b,
+                                                                             context.codemap) => {
+                b.expr.as_ref().map(|e| &**e).unwrap()
```
Comment (nrc) [inline src/expr.rs]: This expression looks odd - can it be simplified?

File: tests/target/match.rs (hunk 3 above)
Comment (nrc) [inline tests/target/match.rs]: I guess we should check that a comment in the block also preserves the block

**Comment by nrc** [comment]
Could we add an option for whether or not we strip/add blocks in match arms? Not sure if it is more important than closures, but it feels like it - maybe because match arms get re-written more often?

r+ with or without an option, but with the nits addressed

File: src/expr.rs (hunk 4 above)
Comment (marcusklaas) [inline src/expr.rs]: I have tried, but this the only expression I could find that did the trick.

**Comment by marcusklaas** [comment]
Added option and additional tests.

**Pull request merged**